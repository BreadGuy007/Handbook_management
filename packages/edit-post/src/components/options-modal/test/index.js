/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { OptionsModal } from '../';

describe( 'OptionsModal', () => {
	it( 'should match snapshot when the modal is active', () => {
		const wrapper = shallow( <OptionsModal isModalActive={ true } /> );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should not render when the modal is not active', () => {
		const wrapper = shallow( <OptionsModal isModalActive={ false } /> );
		expect( wrapper.isEmptyRender() ).toBe( true );
	} );
} );
