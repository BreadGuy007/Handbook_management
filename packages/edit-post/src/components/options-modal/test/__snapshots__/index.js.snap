// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsModal should match snapshot when the modal is active 1`] = `
<WithInstanceId(Modal)
  className="edit-post-options-modal"
  closeLabel="Close"
  title="Options"
>
  <Section
    title="General"
  >
    <WithSelect(WithDispatch(IfViewportMatches(BaseOption)))
      label="Pre-publish checks"
    />
  </Section>
  <Section
    title="Document panels"
  >
    <EnablePluginDocumentSettingPanelOptionSlot />
    <WithSelect(PostTaxonomies)
      taxonomyWrapper={[Function]}
    />
    <PostFeaturedImageCheck>
      <WithSelect(IfCondition(WithDispatch(BaseOption)))
        label="Featured image"
        panelName="featured-image"
      />
    </PostFeaturedImageCheck>
    <PostExcerptCheck>
      <WithSelect(IfCondition(WithDispatch(BaseOption)))
        label="Excerpt"
        panelName="post-excerpt"
      />
    </PostExcerptCheck>
    <WithSelect(PostTypeSupportCheck)
      supportKeys={
        Array [
          "comments",
          "trackbacks",
        ]
      }
    >
      <WithSelect(IfCondition(WithDispatch(BaseOption)))
        label="Discussion"
        panelName="discussion-panel"
      />
    </WithSelect(PostTypeSupportCheck)>
    <WithSelect(PageAttributesCheck)>
      <WithSelect(IfCondition(WithDispatch(BaseOption)))
        label="Page attributes"
        panelName="page-attributes"
      />
    </WithSelect(PageAttributesCheck)>
  </Section>
  <WithSelect(MetaBoxesSection)
    title="Advanced panels"
  />
</WithInstanceId(Modal)>
`;
